### ISSUE TITLE
get_robustcov_results kernel keyword issue

### ISSUE THREAD

**Issue by eph** [statsmodels/statsmodels#4429]
The documentation incorrectly says that the 'kernel' keyword argument in `get_robustcov_results` should be a string instead of a callable.  See here:

http://www.statsmodels.org/devel/generated/statsmodels.regression.linear_model.OLSResults.get_robustcov_results.html?highlight=get_robustcov_results

```python
In [3]: model = sm.OLS(np.random.normal(size=(10,1)), np.random.normal(size=(10,2)))

In [4]: model.fit().get_robustcov_results(cov_type='HAC', kernel='bartlett',maxlags=3)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-64-7041f059add2> in <module>()
----> 1 model.fit().get_robustcov_results(cov_type='HAC', kernel='bartlett',maxlags=3)

~/anaconda3/lib/python3.6/site-packages/statsmodels/regression/linear_model.py in get_robustcov_results(self, cov_type, use_t, **kwds)
   2029             res.cov_params_default = sw.cov_hac_simple(self, nlags=maxlags,
   2030                                                  weights_func=weights_func,
-> 2031                                                  use_correction=use_correction)
   2032         elif cov_type.lower() == 'cluster':
   2033             #cluster robust standard errors, one- or two-way

~/anaconda3/lib/python3.6/site-packages/statsmodels/stats/sandwich_covariance.py in cov_hac_simple(results, nlags, weights_func, use_correction)
    696     '''
    697     xu, hessian_inv = _get_sandwich_arrays(results)
--> 698     sigma = S_hac_simple(xu, nlags=nlags, weights_func=weights_func)
    699 
    700     cov_hac = _HCCM2(hessian_inv, sigma)

~/anaconda3/lib/python3.6/site-packages/statsmodels/stats/sandwich_covariance.py in S_hac_simple(x, nlags, weights_func)
    394         nlags = int(np.floor(4 * (n_periods / 100.)**(2./9.)))
    395 
--> 396     weights = weights_func(nlags)
    397 
    398     S = weights[0] * np.dot(x.T, x)  #weights[0] just for completeness, is 1

TypeError: 'str' object is not callable
```
I am happy make a PR to either fix the documentation (or to change the code to allow for ['bartlett', 'uniform', callable].

**Comment by josef-pkt** [comment]
@eph Thanks for reporting, That is a mistake that needs to be fixed.

as reminder/reference: previous consistency fix for hac/nw was #3032

I would prefer that the code is fixed, and the docstring adjusted to whatever that will be. 
In the medium/long run, we want to add more kernels and more options, and "finding" the kernel functions is not so easy for a user (especially because I use weights_func in the code.). 
So, being able to use a string to choose will be a lot more user friendly.

I think the best would be to define a dict in stats.sandwich_covariance that maps names to functions. That would be easily extentable as we add more weights_xxx kernel functions to the module and dict.

I guess the best place would be add the initial keyword normalization, e.g. after 
https://github.com/statsmodels/statsmodels/blob/master/statsmodels/regression/linear_model.py#L2127

Note: This code is still almost exactly duplicated and needs to be changed twice. Second place is
https://github.com/statsmodels/statsmodels/blob/master/statsmodels/base/covtype.py#L136

**Comment by jbrockmendel** [comment]
Not having duplicated code is also an option.
